src/enumerable.rb

```diff
@@ -528,6 +528,15 @@ def reject
     ary
   end
 
+  def reverse_each(*args)
+    reversed_arr = to_a.reverse
+    return enum_for(:reverse_each, reversed_arr) unless block_given?
```

Comment:
This isn't quite right. The `*args` should be passed in here. I'm not sure why the specs don't test this, but I wrote up a little Ruby program to prove it:

```ruby
class MyEnum
  include Enumerable

  def each(*args)
    args.each do |arg|
      yield arg
    end
  end
end

MyEnum.new.reverse_each(1, 2, 3) do |i|
  p i
end

e = MyEnum.new.reverse_each(1, 2, 3)
e.each do |i|
  p i
end
```

This should print:

```
3
2
1
3
2
1
```

Reply:
Got it, I appreciate you pointing me in the correct directions, the existing specs really led me to take a naive approach.